pkg/operator/resources/asyncapi/api.go

```diff
@@ -64,25 +64,31 @@ func getGatewayK8sName(apiName string) string {
 	return "gateway-" + apiName
 }
 
-func deploymentID() string {
+func generateDeploymentID() string {
 	return k8s.RandomName()[:10]
 }
 
 func UpdateAPI(apiConfig userconfig.API, force bool) (*spec.API, string, error) {
-	prevK8sResources, err := getK8sResources(apiConfig)
+	prevK8sResources, err := getK8sResources(apiConfig.Name)
 	if err != nil {
 		return nil, "", err
 	}
 
-	deployID := deploymentID()
-	if prevK8sResources.apiDeployment != nil && prevK8sResources.apiDeployment.Labels["deploymentID"] != "" {
-		deployID = prevK8sResources.apiDeployment.Labels["deploymentID"]
+	createdTime := time.Now().UnixNano()
+	deploymentID := generateDeploymentID()
+	if prevK8sResources.gatewayVirtualService != nil && prevK8sResources.gatewayVirtualService.Labels["createdTime"] != "" {
```

Comment:
this label is unnecessary. All kubernetes resources have a `creationTimestamp` field

Reply:
The creationTimestamp is required to initialize the SQS queue. The SQS queue is created before the k8s resources are applied. Once there is a CRD for the async api, we wouldn't need to have a separate field like `creationTimestamp`.